src/transform.ts

```diff
@@ -144,8 +161,61 @@ export class Transform {
       throw new Error(`Can't read from segment that has been partially transformed`);
     }
 
-    const sb = shiftBounds(shifts, bounds);
-    return content.slice(sb.start, sb.start + sb.length).toString();
+    return shiftBounds(shifts, bounds);
+  }
+
+  error(node: WithSrc, msg: string): Error {
+    const { source, start } = this.decodeSrc(node.src);
+    const line = byteToLineNumber(this.state[source].originalBuf, start);
+    const error = new Error(`${msg} (${source}:${line})`);
+    Error.captureStackTrace(error, this.error); // capture stack trace without this function
+    return error;
+  }
+
+  getRealEndIndex(node: Node): number {
+    const { source, start, length } = this.decodeSrc(node.src);
+    const buf = this.state[source].originalBuf;
+
+    let end: number | undefined;
+
+    if (node.nodeType !== 'VariableDeclaration') {
+      end = start + length;
+    } else {
+      // VariableDeclaration node bounds don't include the semicolon so we manually look for it
+      for (let i = start + length; i < buf.length; i++) {
+        const c = buf.toString('utf8', i, i + 1);
+        if (c === ';') {
+          end = i;
+          break;
+        } else if (/\S/.test(c)) {
+          throw this.error(node, 'Found unexpected content before semicolon');
+        }
+      }
+      if (end === undefined) {
+        throw this.error(node, 'could not find end of node');
+      }
+    }
+
+    let foundComment = false;
+
+    for (let i = end + 1; i < buf.length; i++) {
+      // look ahead 2 bytes to search for '//'
+      const next = buf.toString('utf8', i, i + 2);
+      if (foundComment) {
+        if (/[^\n\r]/.test(next[0])) {
+          end = i;
+        } else {
+          break;
+        }
+      } else if (next === '//') {
+        end = i + 1;
+        foundComment = true;
+      } else if (/[^ \t]/.test(next[0])) {
+        break;
+      }
+    }
```

Comment:
That works, but its a bit "manual" to me. I'm wondering 2 things:
- Are there `Node` object for the comment in the AST?
- Given a Node, is there a way to get the following one?

If so I think we should:
- get the next Node
  - If it is a comment and if the line where that node starts is the same as the line were the current Node ends (there is no `\n` between them), then we move the end of the node to the end of the comment
  - otherwize, break
- repeate with the following node (in case multiple `/* */` comments are chainned with no newline) ?

Reply:
> Are there Node object for the comment in the AST?
> Given a Node, is there a way to get the following one?

The answer is No to both of these questions :cry:

Reply:
I've changed the way this works. It's now using the `matchBufferAt` helpers that were already present in the codebase. I didn't want to use those because I think they are inefficient (it copies the contents of the buffer), but it's probably ok, the main inefficiencies were around AST dereferencing.